docs/adr/0002-delivery-channel-creation-and-integration.md

```diff
@@ -0,0 +1,81 @@
+
+The management API will host configuration for what notification types can be sent through which delivery channels in the following format:
+
+```json
+[
+  {
+    "id": "",
+    "displayName": "",
+    "communicationTypes": [""],
+    "isConfigured": false
+  }
+]
+```
+where 'communicationTypes' specifies what communication types are supported in this Delivery Channel. This allows for consumers of Marain.UserNotifications to specify if they want to only use one kind communication type from a certain Delivery Channel that supports multiple types.
+
+A delivery channel will have a generic api that can be consumed by the Management Api and be able to transform data provided in the Notification into a format that is specific to that communication type (this will be discussed in a sequential adr). The result of this call will be a callback uri/null depending on how the delivery channel has been implemented.
+
+### Delivery Channel configuration 
+
+A delivery channel will also need the configuration of the underlying platform to be stored somewhere (api keys, secrets, etc). This will be stored in a generic manner so if a supported delivery channel needs to be used, an api key and slight modification of the settings in the management api will only be required. 
+
+Keys for the different platforms in the delivery channels will be stored in a format as follows:
+- `Marain:DeliveryChannel:<Platform_name>:<key_name>`
+- `Marain:DeliveryChannel:Airship:ApiMasterSecret`
+
+The Delivery Channel will also have to have it's own kind of configuration specifying the supported Communication Types. These will be hardcoded values and created when the coding of the new delivery channel is done. 
+
+### Delivery Channel usage
+
+The delivery channel will need to transform an object into the equivalent api model  used to send a message by the chosen platform (twilio, airship, send grid etc). Some notifications will have different headings, bodys, generic templates that will have to be transformed into a format that is suitable and also might have different ways of targetting certain users depending on how the users are registered with the platforms. (for twilio this will be a number, for web push this could be a mixture of business id and email, etc). 
+
+A solution to this would be to create a generic communication type object (smsobject, emailobject, webpushobject) that is provided by the management api to the delivery channel, and then the delivery channel can map a generic communication type object into the required api model. This might prove to be simpler when integrating new delivery channels in the future as most of the information will already be there.
```

Comment:
This is still a bit unclear to me. Is there a plan to have a single data type that represents the information you wish to convey, and for this then to be transformed in to SMS, email, etc (before then being transformed into Twilio's representation of an SMS, SendGrid's representation of an email etc). Or are applications going to need to supply every conceivable kind of communication object?

There are reasons for doing it either way. Emails and SMSs are quite different so there are reasons you might want an application to be able to provide differently-shaped notifications based on the message type. But equally, we might want it to be possible for the application to say "Here's the information I want to convey", and then for the notification service to turn that into an SMS, email, or whatever.

Which approach are we taking?

Reply:
That was unclear, I think I've cleared it up.

Reply:
I'm still not quite clear on what the application is expected to do. Does it need to provide as many different objects as there are communication types? Or is there some all-capable general format we pass in that the management API somehow knows how to map?

I think it might actually be better to have a separate ADR for templating that addresses these concerns.

Reply:
There is now a seperate ADR that covers templating.